### PR TITLE
Add compaction information 

### PR DESCRIPTION
**Pull request by kelseiv**
Closes /influxdata/DAR/issues/42

cc @danatinflux

### CODE REVIEW

File: content/influxdb/v2.2/reference/internals/shards.md [hunk 1]
```diff
@@ -104,25 +104,39 @@ historical data, InfluxDB writes to older shards that must first be un-compacted
...
 ##### Shard compaction-related configuration settings
+
+The following configuration settings are especially beneficial for systems with irregular loads, because they limit compactions during periods of high usage, and let compactions catch up during periods of lower load:
+
 - [`storage-compact-full-write-cold-duration`](/influxdb/v2.2/reference/config-options/#storage-compact-full-write-cold-duration)
 - [`storage-compact-throughput-burst`](/influxdb/v2.2/reference/config-options/#storage-compact-throughput-burst)
 - [`storage-max-concurrent-compactions`](/influxdb/v2.2/reference/config-options/#storage-max-concurrent-compactions)
 - [`storage-max-index-log-file-size`](/influxdb/v2.2/reference/config-options/#storage-max-index-log-file-size)
 - [`storage-series-file-max-concurrent-snapshot-compactions`](/influxdb/v2.2/reference/config-options/#storage-series-file-max-concurrent-snapshot-compactions)
 - [`storage-series-file-max-concurrent-snapshot-compactions`](/influxdb/v2.2/reference/config-options/#storage-series-file-max-concurrent-snapshot-compactions)
```
Comment (kelseiv) [inline content/influxdb/v2.2/reference/internals/shards.md:136]: @davidby-influx, @sanderson, I added some 1.x compaction detail re [DAR](/influxdata/DAR/issues/4) to the 2.x docs. For these specific config settings, is there a 1:1 correlation?

File: content/influxdb/v1.8/concepts/tsi-details.md [hunk 2]
```diff
@@ -86,12 +85,29 @@ The following occurs when a write comes into the system:
...
+
+Once the `LogFile` exceeds a threshold (`5MB`), InfluxDB creates a new active log file, and the previous one begins compacting into an `IndexFile`. This first index file is at level 1 (L1). The log file is considered level 0 (L0). Index files can also be created by merging two smaller index files together.
+For example, if two contiguous L1 index files exist, InfluxDB merges them into an L2 index file.
+
+InfluxDB schedules compactions preferentially, using the following guidelines:
+
+- The lower the level (the fewer times a file has been compacted), the more weight is given to compacting it.
+- The more compactible files in a level, the higher the priority given to that level. If the number of files in each level is equal, lower levels are compacted first.
+- If a higher level has more candidates for compaction, it may be compacted before a lower level. The equation is the number of candidate files for compaction times level weights of 0.4, 0.3, 0.2, and 0.1.
```
Comment (kelseiv) [inline content/influxdb/v1.8/concepts/tsi-details.md:97]: @davidby-influx the equation to calculate the compaction priority?....
...the weight is multiplied by the number of compactible files in any given level --no matter the level?

File: content/influxdb/v2.2/reference/internals/shards.md (hunk 1 above)
Comment (davidby-influx) [reply content/influxdb/v2.2/reference/internals/shards.md:136]: Yes, there is.    Here is the equivalence table:

https://github.com/influxdata/influxdb/blob/8c9768cdb74a458948714f988ecd7b96a14fa3da/cmd/influxd/upgrade/config.go#L20

**Review by sanderson — APPROVED** [review]
Looks good.

**Pull request merged**